doc/neps/nep-0051-scalar-representation.rst

```diff
@@ -0,0 +1,332 @@
+
+An exception to this are downstream libraries with documentation and
+especially documentation testing.
+Since the representation of many values will change, in many cases
+the documentation will have to be updated.
+This is expected to require larger documentation fixups in the mid-term.
+
+Further, it may be necessary to adept tools for doctest testing to
+allow approximate value checking for the new representation.
+
+Changes to ``arr.tofile()``
+---------------------------
+``arr.tofile()`` currently stores values as ``repr(arr.item())`` when in text
+mode.  This is not always ideal also since that may include a conversion to
+Python.
+One issue is that this would start saving longdouble as
+``np.longdouble('3.1')`` which is clearly not desired.  We expect that this
+method is rarely used for object arrays.  For string arrays, using the ``repr``
+also leads to storing ``"string"`` or ``b"string"`` which seems rarely desired.
+
+The proposal is to change the default (back) to use ``str`` rather than
+``repr``.  If ``repr`` is desired, users will have to pass ``fmt=%r``.
+
+
+Detailed description
+====================
+
+This NEP proposes to change the represenatation for NumPy scalars to:
+
+* ``np.True_`` and ``np.False_`` for booleans
+* ``np.scalar(<value>)``, i.e. ``np.float64(3.0)`` for all numerical dtypes.
+* The value for ``np.longdouble`` and ``np.clongdouble`` will be given in quotes:
+  ``np.longdouble('3.0')``.  This ensures that it can always roundtrip correctly
+  and matches the way that ``decimal.Decimal`` behaves.
+  For these two the size based name such as ``float128`` will not be used
+  as it is platform dependend and misleading.
+* ``np.str_("string")`` and ``np.bytes_(b"byte_string")`` for string dtypes.
+* ``np.void((3, 5), dtype=[('a', '<i8'), ('b', 'u1')])`` (similar to arrays),
+  this will be valid syntax to recreate the scalar.
+  Unlike arrays, the representation should round-trip correctly, so longdouble
+  values will be quoted and other values never be truncated.
+
+Where booleans are printed as their singletons since this is more concise.
+For strings we include the ``np.`` as ``str_`` and ``bytes_`` on their
+own may not be sufficient to indicate NumPy involvement.
+
+Affects on Masked Arrays
+------------------------
+Some other parts of NumPy may indirectly be changed here.  Masked arrays
+``fill_value`` will be adapted to only include the full scalar information
+such as ``fill_value = np.float64(1e20)`` when the dtype of the array
+mismatches.
+For longdouble (with matching dtype), it will be printed as
+``fill_value='3.1'`` similar to 
+
+Affect on records
+-----------------
+
+The ``np.record`` scalar will be aligned with ``np.void`` and print identically
+to it (except the name itself).
+
+New public API
+--------------
+
+Void scalars and the masked array ``fill_value`` require access to printing
+the scalar value as ``'3.1'`` rather than ``np.longdouble('3.1')``, or for
+strings
```

Comment:
The meaning here isn't clear to me - it seems like these three sections are interrelated and might benefit from a high-level reorganization to improve the flow & clarity.

Reply:
Merging MA and record into a single section, and removing this last bit.  Added a paragraph about the need for `get_formatter() above though, maybe you can have a look over that/whether it makes sense to you.